cloud/scope/route_table_reconciler.go

```diff
@@ -89,23 +89,25 @@ func (s *ClusterScope) getRouteTable(ctx context.Context, routeTableType string)
 		}
 
 	}
-	vcId := s.getVcnId()
+	vcnId := s.getVcnId()
 	routeTableName := PublicRouteTableName
 	if routeTableType == infrastructurev1beta2.Private {
 		routeTableName = PrivateRouteTableName
 	}
-	rts, err := s.VCNClient.ListRouteTables(ctx, core.ListRouteTablesRequest{
-		CompartmentId: common.String(s.GetCompartmentId()),
-		VcnId:         vcId,
-		DisplayName:   common.String(routeTableName),
-	})
-	if err != nil {
-		s.Logger.Error(err, "failed to list route tables")
-		return nil, errors.Wrap(err, "failed to list route tables")
-	}
-	for _, rt := range rts.Items {
-		if s.IsResourceCreatedByClusterAPI(rt.FreeformTags) {
-			return &rt, nil
+	if vcnId != nil {
```

Comment:
By doing this are we swallowing errors? If, for whatever reason, the `vcnId` is `nil` we won't return an error to a user. Maybe we should leave that in. IDK